aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -490,6 +498,36 @@ module aptos_token::token {
         property_map::update_property_map(value, keys, values, types);
     }
 
+    // Update token uri if mutable
+    public entry fun update_token_uri(
+        account: &signer,
+        creator_address: address,
+        token_id: TokenId,
+        new_uri: String
+    ) acquires Collections, TokenStore {
+
+        assert!(signer::address_of(account) == creator_address, error::not_found(ENO_MUTATE_CAPABILITY));
```

Comment:
We have this same redundancy issue in another method as well (but too late to change the signature of that one now as it's not backward compatible). Here, since this is a new function, can we remove the creator_address argument? If account is expected to be the creator anyway, creator address can just be extracted from the account's address.

Reply:
Yeah good catch! I'll work on it tomorrow 👍

Reply:
Let's make a public function first to make it usable by other contracts.  You can use the ceator in the TokenID.TokenDataID to get the creator address of the token

Reply:
let's call is mutate_token_uri to make it consistent with other naming

Reply:
please name the function `mutate_tokendata_uri`